bot/worker.py

```diff
@@ -0,0 +1,182 @@
+from util.logger import logger
+from util.str_utils import preprocess_text
+
+__author__ = 'Jonarzz'
+__maintainer__ = 'MePsyDuck'
+
+cache_api = get_cache_api()
+
+
+def work():
+    """Main method executing the script.
+
+    It connects to an account, loads dictionaries from proper files (declared in config file).
+    Afterwards it executes process_comments method with proper arguments passed.
+    """
+
+    reddit = account.get_account()
+    logger.info('Connected to Reddit account : ' + config.USERNAME)
+
+    comment_stream = reddit.subreddit(config.SUBREDDIT).stream.comments(pause_after=-1)
+    submission_stream = reddit.subreddit(config.SUBREDDIT).stream.submissions(pause_after=-1)
+    while True:
+        for comment in comment_stream:
+            if comment is None:
+                break
+            process_replyable(reddit, comment)
+        for submission in submission_stream:
+            if submission is None:
+                break
+            process_replyable(reddit, submission)
+
+
+def process_replyable(reddit, replyable):
+    """Method used to check all the comments in a submission and add replies if they are responses.
+
+    PRAW generates past ~100 comments/submissions on the first iteration. Then the loop only runs if there is a new
+    comment/submission added to the stream. This also means that once PRAW is up and running, after the initial comments
+    list it won't generate any duplicate comments.
+
+    However, just as a safeguard, Caching is used to store replyable ids as they are processed for the first time.
+    Otherwise, when the bot is restarted it might reply twice to same comments. If replyable id is in the already present
+    in the cache_api, then it is ignored, else processed and added to the cache_api.
+    * Self comments are ignored.
+    * It is prepared for comparision to the responses in dictionary.
+    * If the replyable is not on the excluded responses list (loaded from config) and if it is in the responses db or
+    specific responses list, a reply replyable is prepared and posted.
+
+    :param reddit: The reddit account instance
+    :param replyable: comment or submission
+    :return: None
+    """
+
+    if cache_api.check(thing_id=replyable.fullname):
+        return
+
+    # Ignore thyself
+    if replyable.author == reddit.user.me:
+        return
+
+    logger.debug("Found new replyable: " + str(replyable.fullname))
+
+    processed_body = process_body(replyable.body if isinstance(replyable, Comment) else replyable.title)
+
+    # Don't reply to single word text (they're mostly common phrases).
+    if ' ' not in processed_body:
+        return
+
+    if processed_body in config.EXCLUDED_RESPONSES:
+        return
+
+    if processed_body in config.CUSTOM_RESPONSES:
+        do_custom_reply(replyable=replyable, custom_response=config.CUSTOM_RESPONSES[processed_body])
+
+    if try_flair_specific_reply(replyable, processed_body):
+        return
+    else:
```

Comment:
Obsolete else after the early return.

Reply:
But isn't the purpose of `if` here try to reply with flair specific reply, else reply with general reply? `return` would only happen if the comment/submisson was already replied to with flair specific response reply.

Reply:
So maybe something like

        if not try_flair_specific_reply(replyable, processed_body):
            do_regular_reply(replyable, processed_body)

yet chaning `try_flair_specific_reply` name could increase readibility.

Reply:
Refactored and changed the code to be more readable in e3c93c18a268766a735276fa24053ae5059aa18e.